app/components/settings_drawer/index.js

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+
+import {logout} from 'mattermost-redux/actions/users';
+import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+
+import SettingsDrawer from './settings_drawer';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+
+function mapStateToProps(state) {
+    return {
+        currentUser: getCurrentUser(state) || {},
```

Comment:
Could you use defaultProps to set that to {}? I know it'll be very rare for this to be undefined, but I think that's better form since it won't make a new object each time

Reply:
Can do